students/KevinWaldron/lesson04/mailroom.py

```diff
@@ -0,0 +1,104 @@
+# (for easy sorting), followed by the history of donations
+donors = {
+"Bill Ted":[726.63, 353.53, 348.1, 25.00],
+"Frank Fred":[178.76, 120.50, 56.76, 1.50],
+"Laura Todd":[5.75, 5.75],
+"Steve Lincoln":[165.28, 75.38, 89.9],
+"Lisa Grant":[209.7, 175.50, 34.20]
+}
+
+def list_donors():
+    """ Prints a list of the donors by name """
+    print("\n" + ("-"*10) + " Donors " + ("-"*10))
+    for donor in donors:
+        print(donor)
+
+def quit_menu():
+    """ Handles the menu quit selection """
+    return "Quit"
+
+def create_thank_you(donation):
+    """
+    Prints a thank for you the given donor and amount
+    :param donation: dictionary containing name of the donor to thank and the amount donated
+    :returns: formatted thank you string
+    """
+    return f"""Dear {donation["name"]},\nThank you for your very generous donation of ${donation["amount"]:.2f}.  It \nwill go very far in supporting the Human Fund, \"Money for \nPeople.\"\n{"Sincerely":>40}\n{"Art Vandelay":>50}"""
+
+def donor_entry():
+    """ Handles donor entry menu selection """
+    name = input("\nEnter donor's full name or 'q' to return to the previous menu: ")
+    if(name == "q"):
+        return
+    amount = input("Enter donation amount or 'q' to return to the main menu: ")
+    if (amount == "q"):
+        return
+    if(not name in donors):
+        donors[name] = [0]
+    amount = float(amount)
+    donors[name].append(amount)
+    donors[name][0] += amount
+    print(donors[name])
+    print("\n" + create_thank_you({"name":name, "amount":amount}))
+
+# modify donor entry slightly (no more list) to reuse display prompt funtion
+thank_you_menu_prompt = "1. Enter donation\n2. See a list of donors\n3. Return to previous menu"
+thank_you_menu_dict = {'1':donor_entry, '2':list_donors, '3':quit_menu}
+
+def send_thank_you():
+    # modify donor entry slightly (no more list) to reuse display prompt funtion
+    thank_you_menu_prompt = "1. Enter donation\n2. See a list of donors\n3. Return to previous menu"
+    thank_you_menu_dict = {'1':donor_entry, '2':list_donors, '3':quit_menu}
+    """ Handles the send thank you menu selection """
+    display_prompt(thank_you_menu_prompt, thank_you_menu_dict)
+
+def total_donations(donor):
+    """ Key sort method to retrieve total contribution amount for donors data structure """
+    return donor[1]
+
+def mail_everyone():
+    """ Handles the mail to everyone menu selection """
+    for donor in donors:
+        with open(donor.lower().replace(' ', "_") + ".txt", 'w') as f:
+            f.write(create_thank_you({"name":donor, "amount":donors[donor][0]}))
+
+def create_report():
+    """ Handles the create report menu selection """
+    header = "\n{:<20}| Total Given | Num Gifts | Average Gift".format("Donor Name")
+    print(header)
+    print("-" * (len(header) - 1))
+    # pack the donors into a list of lists for soting
+    donors_list = []
+    for name in donors:
```

Comment:
more pythonic to loop over keys and values in dictionary if you are using both:
`for name, donations in donors.items()`